pkg/backend/openshiftcluster/create.go

```diff
@@ -41,7 +44,26 @@ func (m *Manager) Create(ctx context.Context) error {
 
 	resourceGroup := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
 
+	pullSecret := os.Getenv("PULL_SECRET")
 	m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
+		// Ensure the password is encrypted at rest by using the SecureBytes or SecureString type.
+		if m.env.GenerateACRToken() {
+			// this should only happen on prod only
+			acrTokenName := uuid.NewV4().String()
+			acrPassword := 	 uuid.NewV4().String()
+
+			err = acrtoken.CreateACRToken(ctx, m.env, acrTokenName, acrPassword)
+			if err != nil {
+				return err
+			}
+			pullSecret, err = pullsecret.ReplaceRegistryAuth(pullSecret, "arosvc.azurecr.io", acrTokenName, acrPassword)
+			if err != nil {
+				return err
+			}
```

Comment:
pull this bit out of the PatchWithLease.  Confirm that CreateACRToken will not error if called twice with the same arguments (e.g. if the RP crashes)